src/goto-symex/symex_builtin_functions.cpp

```diff
@@ -530,37 +518,33 @@ void goto_symext::symex_macro(
 {
   const irep_idt &identifier=code.op0().get(ID_identifier);
 
-  if(identifier==CPROVER_MACRO_PREFIX "waitfor")
-  {
-    #if 0
-    exprt new_fc("waitfor", fc.type());
+  PRECONDITION(identifier == CPROVER_MACRO_PREFIX "waitfor")
+  #if 0
+  exprt new_fc("waitfor", fc.type());
 
-    if(fc.operands().size()!=4)
-      throw "waitfor expected to have four operands";
+  if(fc.operands().size()!=4)
+    throw "waitfor expected to have four operands";
 
-    exprt &cycle_var=fc.op1();
-    exprt &bound=fc.op2();
-    exprt &predicate=fc.op3();
+  exprt &cycle_var=fc.op1();
+  exprt &bound=fc.op2();
+  exprt &predicate=fc.op3();
 
-    if(cycle_var.id()!=ID_symbol)
-      throw "waitfor expects symbol as first operand but got "+
-            cycle_var.id();
+  if(cycle_var.id()!=ID_symbol)
+    throw "waitfor expects symbol as first operand but got "+
+          cycle_var.id();
 
-    exprt new_cycle_var(cycle_var);
-    new_cycle_var.id("waitfor_symbol");
-    new_cycle_var.copy_to_operands(bound);
+  exprt new_cycle_var(cycle_var);
+  new_cycle_var.id("waitfor_symbol");
+  new_cycle_var.copy_to_operands(bound);
 
-    replace_expr(cycle_var, new_cycle_var, predicate);
+  replace_expr(cycle_var, new_cycle_var, predicate);
 
-    new_fc.operands().resize(4);
-    new_fc.op0().swap(cycle_var);
-    new_fc.op1().swap(new_cycle_var);
-    new_fc.op2().swap(bound);
-    new_fc.op3().swap(predicate);
+  new_fc.operands().resize(4);
+  new_fc.op0().swap(cycle_var);
+  new_fc.op1().swap(new_cycle_var);
+  new_fc.op2().swap(bound);
+  new_fc.op3().swap(predicate);
 
-    fc.swap(new_fc);
-    #endif
-  }
-  else
-    throw "unknown macro: "+id2string(identifier);
```

Comment:
Here, I consider the whole 

```
if (something)
{
   ...some computation...
} else {
   throw exception
}
```

to just have the general form:

```
assert (something)
... do the useful computation ...
```

so I have rearranged it appropriately.

Reply:
That's assuming that (something) is error checked elsewhere, and that its not being provided from user code....